rasa/core/actions/forms.py

```diff
@@ -118,44 +146,50 @@ def intent_is_desired(
 
     def entity_is_desired(
         self,
-        requested_slot_mapping: Dict[Text, Any],
+        slot_mapping: Dict[Text, Any],
         slot: Text,
         entity_type_of_slot_to_fill: Optional[Text],
-        tracker: "DialogueStateTracker",
+        tracker: DialogueStateTracker,
+        domain: Domain,
     ) -> bool:
         """Check whether slot should be filled by an entity in the input or not.
 
         Args:
-            requested_slot_mapping: Slot mapping.
+            slot_mapping: Slot mapping.
             slot: The slot to be filled.
             entity_type_of_slot_to_fill: Entity type of slot to fill.
             tracker: The tracker.
+            domain: The domain.
 
         Returns:
             True, if slot should be filled, false otherwise.
         """
 
         # slot name is equal to the entity type
-        slot_equals_entity = slot == requested_slot_mapping.get("entity")
+        slot_equals_entity = slot == slot_mapping.get("entity")
+        entity_mapping_is_unique = self._entity_mapping_is_unique(slot_mapping, domain)
 
         # use the custom slot mapping 'from_entity' defined by the user to check
         # whether we can fill a slot with an entity (only if a role or a group label
         # is set)
         if (
-            requested_slot_mapping.get("role") is None
-            and requested_slot_mapping.get("group") is None
-        ) or entity_type_of_slot_to_fill != requested_slot_mapping.get("entity"):
+            slot_mapping.get("role") is None and slot_mapping.get("group") is None
+        ) or entity_type_of_slot_to_fill != slot_mapping.get("entity"):
             slot_fulfils_entity_mapping = False
         else:
             matching_values = self.get_entity_value(
-                requested_slot_mapping.get("entity"),
+                slot_mapping.get("entity"),
                 tracker,
-                requested_slot_mapping.get("role"),
-                requested_slot_mapping.get("group"),
+                slot_mapping.get("role"),
+                slot_mapping.get("group"),
             )
             slot_fulfils_entity_mapping = matching_values is not None
 
-        return slot_equals_entity or slot_fulfils_entity_mapping
+        return (
+            slot_equals_entity
+            or entity_mapping_is_unique
```

Comment:
Why do we add this condition? Can you please explain and add a comment? Thanks.

Reply:
added comment